runtime/common/src/adapters.rs

```diff
@@ -161,3 +163,29 @@ where
 		pallet_ema_oracle::Pallet::<Runtime>::get_price_weight()
 	}
 }
+
+pub struct PriceAdjustmentAdapter<Runtime, LMInstance>(PhantomData<(Runtime, LMInstance)>);
+
+impl<Runtime, LMInstance> PriceAdjustment<GlobalFarmData<Runtime, LMInstance>>
+	for PriceAdjustmentAdapter<Runtime, LMInstance>
+where
+	Runtime: warehouse_liquidity_mining::Config<LMInstance>
+		+ pallet_ema_oracle::Config
+		+ pallet_omnipool_liquidity_mining::Config,
+	u32: From<<Runtime as warehouse_liquidity_mining::Config<LMInstance>>::AssetId>,
```

Comment:
why this is needed ? looks bit out of place;-)

Reply:
Because oracle is using `u32` as AssetId and LM has it as associated type without any restriction to u32. 
https://github.com/galacticcouncil/warehouse/blob/edbff0b40145fe0afd124fde9be3a1962c27fade/price-oracle/src/types.rs#L29
https://github.com/galacticcouncil/warehouse/blob/edbff0b40145fe0afd124fde9be3a1962c27fade/liquidity-mining/src/lib.rs#L176
You requested some changes in the warehouse PR so I can change it in the warehouse and this doesn't have to be here